parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -419,17 +433,18 @@ public ReadBuilder recordsPerBatch(int numRowsPerBatch) {
         ParquetReadOptions options = optionsBuilder.build();
 
         if (batchedReaderFunc != null) {
-          return new VectorizedParquetReader(file, schema, options, batchedReaderFunc, filter, reuseContainers,
-              caseSensitive, maxRecordsPerBatch);
+          return new VectorizedParquetReader(file, schema, options, batchedReaderFunc, nameMapping,
+              applyNameMapping, filter, reuseContainers, caseSensitive, maxRecordsPerBatch);
         } else {
           return new org.apache.iceberg.parquet.ParquetReader<>(
-              file, schema, options, readerFunc, filter, reuseContainers, caseSensitive);
+              file, schema, options, readerFunc, nameMapping, applyNameMapping, filter, reuseContainers,
+              caseSensitive);
         }
       }
 
       ParquetReadBuilder<D> builder = new ParquetReadBuilder<>(ParquetIO.file(file));
 
-      builder.project(schema);
+      builder.project(schema).withNameMapping(MappingUtil.create(schema));
```

Comment:
Parquet should never automatically add a mapping. When reading, a name mapping should be parsed and added if one is present in table properties.

Reply:
Agreed.